.github/workflows/check-pr-title.yaml

```
@@ -0,0 +1,31 @@
+name: "Check PR title"
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+
+jobs:
+  check-title:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if the PR title is well dressed
+        env:
+          PR_TITLE: ${{ github.event.pull_request.title }}
+          CONV: '(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(\([\w\-\.]+\))?!?'
+          ROX: '(ROX-[0-9]+,)*ROX-[0-9]+'
+          TEXT: ': .+'
+          LENGTH: '70'
+        run: |
+          # Either conventional or ROX-nnn prefix followed by ': ' and random text:
+          REGEX="(($CONV)|($ROX))($TEXT)"
+          if ! printf %s "$PR_TITLE" | grep --perl-regexp --line-regexp "$REGEX"
+          then
+            echo "::error::The PR title doesn't follow the convention."
```

1. Is there a special meaning of `::error::`?
2. You can output this message to `stderr` which is fancier than `stdout` for such a thing.
3. Let's change the message to motivate instead of punishing.
4. As a nice-to-have, you can provide a few examples that comply with the convention. More than that, you can have a self-test of `$REGEX` as part of this same step. That will make sure our rules don't diverge from what we should see.

```suggestion
            >&2 echo "::error::Please update the PR title so that it follows the convention."
```

---

1. Yes, this github syntax for job error output (see [workflow commands](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message));
2. See 1;
3. Ok, let's punish later;
4. Good idea!